frontends/ol-components/src/components/LearningResourceExpanded/DifferingRunsTable.tsx

```diff
@@ -0,0 +1,138 @@
+  display: "flex",
+  flexWrap: "wrap",
+  alignItems: "center",
+  gap: "16px",
+  padding: "12px",
+  alignSelf: "stretch",
+  borderBottom: `1px solid ${theme.custom.colors.lightGray2}`,
+})
+
+const DifferingRunHeader = styled.div({
+  display: "flex",
+  alignSelf: "stretch",
+  alignItems: "center",
+  flex: "1 0 0",
+  gap: "16px",
+  padding: "12px",
+  color: theme.custom.colors.darkGray2,
+  backgroundColor: theme.custom.colors.lightGray1,
+  ...theme.typography.subtitle3,
+})
+
+const DifferingRunData = styled.div({
+  display: "flex",
+  flexShrink: 0,
+  flex: "1 0 0",
+  color: theme.custom.colors.darkGray2,
+  ...theme.typography.body3,
+})
+
+const DifferingRunLabel = styled.strong({
+  display: "flex",
+  flex: "1 0 0",
+})
+
+const DifferingRunLocation = styled(DifferingRunData)({
+  flex: "1 0 100%",
+  flexDirection: "column",
+  alignSelf: "stretch",
+})
+
+const DifferingRunsTable: React.FC<{ resource: LearningResource }> = ({
+  resource,
+}) => {
+  if (!resource.runs) {
+    return null
+  }
+  if (resource.runs.length === 1) {
+    return null
+  }
+  const asTaughtIn = resource ? showStartAnytime(resource) : false
+  const prices: LearningResourcePrice[] = []
+  const deliveryMethods = []
+  const locations = []
+  for (const run of resource.runs) {
+    if (run.resource_prices) {
+      run.resource_prices.forEach((price) => {
+        if (price.amount !== "0") {
+          prices.push(price)
+        }
+      })
+    }
+    if (run.delivery) {
+      deliveryMethods.push(run.delivery)
+    }
+    if (run.location) {
+      locations.push(run.location)
+    }
+  }
+  const distinctPrices = [...new Set(prices.map((p) => p.amount).flat())]
+  const distinctDeliveryMethods = [
+    ...new Set(deliveryMethods.flat().map((dm) => dm?.code)),
+  ]
+  const distinctLocations = [...new Set(locations.flat().map((l) => l))]
+  if (
+    distinctPrices.length > 1 ||
+    distinctDeliveryMethods.length > 1 ||
+    distinctLocations.length > 1
+  ) {
+    return (
+      <DifferingRuns data-testid="differing-runs-table">
+        <DifferingRunHeader>
+          <DifferingRunLabel>Date</DifferingRunLabel>
+          <DifferingRunLabel>Price</DifferingRunLabel>
+          <DifferingRunLabel>Format</DifferingRunLabel>
+        </DifferingRunHeader>
+        {resource.runs.map((run, index) => (
+          <DifferingRun key={index}>
+            <DifferingRunData>
+              {formatRunDate(run, asTaughtIn)}
+            </DifferingRunData>
+            {run.resource_prices && (
+              <DifferingRunData>
+                <span>{getDisplayPrice(getRunPrices(run)["course"])}</span>
+              </DifferingRunData>
+            )}
+            {run.delivery && (
+              <DifferingRunData>
+                <span>{run.delivery?.map((dm) => dm?.name).join(", ")}</span>
+              </DifferingRunData>
+            )}
+            {run.delivery.filter((d) => d.code === "in_person").length > 0 &&
```

Comment:
Is `run.delivery` always an array here as there are conditionals on it above?

Reply:
Yea, according to the API spec it should always be an array. I've also seen it return an array with a single element:

```ts
delivery: Array<CourseResourceDeliveryInner>
```